tidy3d/plugins/autograd/invdes/filters.py

```diff
@@ -1,71 +1,226 @@
-from functools import partial
-from typing import Tuple, Union
+import abc
+from functools import lru_cache, partial
+from typing import Callable, Iterable, Tuple, Union
 
 import numpy as np
+import pydantic.v1 as pd
+from numpy.typing import NDArray
+
+from tidy3d.components.base import Tidy3dBaseModel
 
 from ..functions import convolve
 from ..types import KernelType, PaddingType
-from ..utilities import make_kernel
+from ..utilities import get_kernel_size_px, make_kernel
+
+
+class AbstractFilter(Tidy3dBaseModel, abc.ABC):
+    """A filter class for creating and applying convolution filters.
+
+    Parameters
+    ----------
+    kernel_size : Tuple[pd.PositiveInt, ...]
```

Comment:
docstring should reflect single int case

Reply:
actually, we dont need this parameters list in the docstring do we?

Reply:
ooh yeah i see. i'll remove those then!